packages/ui/src/multiplying-architecture/KaotoEditorChannelApi.ts

```diff
@@ -12,4 +12,32 @@ export interface KaotoEditorChannelApi extends KogitoEditorChannelApi {
    * Returns the Kaoto VSCode settings defined.
    */
   getVSCodeKaotoSettings(): Promise<ISettingsModel>;
+
+  /**
+   * Get metadata querying a Kaoto metadata file.
+   * @param path The path of the file being edited
+   * @param key The key to retrieve the preferences
+   */
+  getMetadata<T>(path: string, key: string): Promise<T>;
```

Comment:
shouldn't it be able to return undefined in case there is no metadata defined?

`getMetadata<T>(path: string, key: string): Promise<T | undefined>;`